test/test_model.py

```diff
@@ -1355,15 +1457,22 @@ def setUp(self):
               bar:
                 k: v
         ''')
-        self.peer_rel_id = self.harness.add_relation('db2', 'db2')
-        self.app = self.harness.model.app
-        self.addCleanup(self.harness.cleanup)
+        yield harness
+        harness.cleanup()
+
+    @pytest.fixture
+    def app(self, harness: ops.testing.Harness[ops.CharmBase]):
+        return harness.model.app
+
+    @pytest.fixture
+    def peer_rel_id(self, harness: ops.testing.Harness[ops.CharmBase]):
+        return harness.add_relation('db2', 'db2')
```

Comment:
Similar here -- unless I'm missing something, these definitely shouldn't be fixtures. Just do `harness.model.app` and `harness.add_relation()` directly in the test. It will make the test more explicit and easier to understand.